src/test/java/reactor/netty/tcp/TcpServerTests.java

```diff
@@ -934,6 +936,77 @@ public void testGracefulShutdown() throws Exception {
 		            .verifyComplete();
 	}
 
+	@Test(expected = ChannelBindException.class)
+	public void testTcpServerWithDomainSocketsNIOTransport() {
+		LoopResources loop = LoopResources.create("testTcpServerWithDomainSocketsNIOTransport");
+		try {
+			TcpServer.create()
+			         .runOn(loop, false)
+			         .bindAddress(() -> new DomainSocketAddress("/tmp/test.sock"))
+			         .bindNow();
+		}
+		finally {
+			loop.disposeLater()
+			    .block(Duration.ofSeconds(30));
+		}
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testTcpServerWithDomainSocketsWithHost() {
+		TcpServer.create()
+		         .bindAddress(() -> new DomainSocketAddress("/tmp/test.sock"))
+		         .host("localhost")
+		         .bindNow();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testTcpServerWithDomainSocketsWithPort() {
+		TcpServer.create()
+		         .bindAddress(() -> new DomainSocketAddress("/tmp/test.sock"))
+		         .port(1234)
+		         .bindNow();
+	}
+
+	@Test
+	public void testTcpServerWithDomainSockets() throws Exception {
+		if (!LoopResources.hasNativeSupport()) {
+			return;
+		}
+		DisposableServer disposableServer =
+				TcpServer.create()
+				         .bindAddress(() -> new DomainSocketAddress("/tmp/test.sock"))
+				         .wiretap(true)
+				         .handle((in, out) -> out.send(in.receive().retain()))
+				         .bindNow();
+
+		Connection conn =
+				TcpClient.create()
+				         .remoteAddress(() -> new DomainSocketAddress("/tmp/test.sock"))
+				         .wiretap(true)
+				         .connectNow();
+
+		conn.outbound()
+		    .sendString(Flux.just("1", "2", "3"))
+		    .then()
+		    .subscribe();
+
+		CountDownLatch latch = new CountDownLatch(1);
+		conn.inbound()
+		    .receive()
+		    .asString()
+		    .doOnNext(s -> {
+		        if (s.endsWith("3")) {
```

Comment:
Why `endsWith` and not `equals`?

Reply:
There is no guarantee how the messages will be received on the server side, this depends on the buffers on network level and the flushes that will be performed.


For example here with this test
The client sends
```
16:18:28.996 [reactor-tcp-kqueue-2] DEBUG reactor.netty.tcp.TcpClient - [id: 0xe6610674, L:null - R:/tmp/test.sock] WRITE: 1B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 31                                              |1               |
+--------+-------------------------------------------------+----------------+
16:18:28.998 [reactor-tcp-kqueue-2] DEBUG reactor.netty.tcp.TcpClient - [id: 0xe6610674, L:null - R:/tmp/test.sock] WRITE: 1B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 32                                              |2               |
+--------+-------------------------------------------------+----------------+
16:18:28.999 [reactor-tcp-kqueue-2] DEBUG reactor.netty.tcp.TcpClient - [id: 0xe6610674, L:null - R:/tmp/test.sock] WRITE: 1B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 33                                              |3               |
+--------+-------------------------------------------------+----------------+
```

on the server this is the incoming and outgoing data:

```
16:18:29.002 [reactor-tcp-kqueue-3] DEBUG reactor.netty.tcp.TcpServer - [id: 0x2b5371dd] READ: 3B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 31 32 33                                        |123             |
+--------+-------------------------------------------------+----------------+
16:18:29.004 [reactor-tcp-kqueue-3] DEBUG reactor.netty.tcp.TcpServer - [id: 0x2b5371dd] WRITE: 3B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 31 32 33                                        |123             |
+--------+-------------------------------------------------+----------------+
```